packages/components/src/input-control/input-field.tsx

```diff
@@ -193,10 +207,13 @@ function InputField(
 	 */
 	let handleOnMouseDown;
 	if ( type === 'number' ) {
-		handleOnMouseDown = ( event ) => {
+		handleOnMouseDown = ( event: MouseEvent< HTMLInputElement > ) => {
 			props.onMouseDown?.( event );
-			if ( event.target !== event.target.ownerDocument.activeElement ) {
-				event.target.focus();
+			if (
+				event.target !==
+				( event.target as HTMLInputElement ).ownerDocument.activeElement
+			) {
+				( event.target as HTMLInputElement ).focus();
```

Comment:
I guess we can make it more readable:
```suggestion
			const target = event.target as HTMLInputElement;
			if ( target !== target.ownerDocument.activeElement ) {
				target.focus();
```
Alternatively, we can just point to `event.currentTarget` instead (it will inherit the type you passed to `MouseEvent`:
```suggestion
			if ( event.currentTarget !== event.currentTarget.ownerDocument.activeElement ) {
				event.currentTarget.focus();
```

Reply:
Ohhhh TIL about current target!

Reply:
Just double-checking — since `target` and `currentTarget` are not guaranteed to be the same HTML element, is this change safe to apply?

I guess it probably is, given that the events are attached to the `input` element (which can't have a descendant element), but I thought it was better to make sure.